opencga-storage/opencga-storage-mongodb/src/main/java/org/opencb/opencga/storage/mongodb/variant/VariantMongoDBIterator.java

```diff
@@ -10,15 +11,15 @@
  */
 public class VariantMongoDBIterator extends VariantDBIterator {
 
-    private FindIterable<Document> dbCursor;
+    private MongoCursor<Document> dbCursor;
     private DocumentToVariantConverter documentToVariantConverter;
 
     VariantMongoDBIterator(FindIterable<Document> dbCursor, DocumentToVariantConverter documentToVariantConverter) { //Package protected
```

Comment:
Should `FindIterable<Document>` be changed to `MongoCursor<Document>` as well?

Reply:
To confine the iterator state within the VariantMongoDBIterator (and not pass it in externally), I passed in a FindIterable (not the cursor itself) to the constructor and then used it to instantiate the iterator inside of the constructor. I admit it is a bit awkward but there is a guarantee that a new iterator is created every time this object is instantiated.